File: Domovoy/Models/Informer/ActiveSession.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Domovoy.Models;

public class ActiveSession
{
    public int Id { get; set; }
    public EventInformer EventInformer { get; set; }
    public int EventInformerId { get; set; }

    public DateTime Start { get; set; }
    public DateTime End { get; set; }
    public string Comment { get; set; }

    [NotMapped] public TimeSpan Duration => End - Start;
}

File: Domovoy/ClientApp/src/api/models/ActiveSession.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { EventInformer } from './EventInformer';

export type ActiveSession = {
    id?: number;
    eventInformer?: EventInformer;
    eventInformerId?: number;
    start?: string;
    end?: string;
    comment?: string | null;
    readonly duration?: string;
};